src/pudl/output/ferc1.py

```diff
@@ -3044,25 +3098,109 @@ def out_ferc1__yearly_rate_base(
         # the assets/liabilites both use ending_balance for its main $$ column
         .rename(columns={"dollar_value": "ending_balance"})
     )
-    # then select only the leafy exploded records that are in rate base and concat
-    in_rate_base = pd.concat(
+    return cash_working_capital
+
+
+def disaggregate_unlabeled_or_total_tag(
+    rate_base_df: pd.DataFrame,
+    tag_col: str,
+) -> pd.DataFrame:
+    """Disaggregate records with an null or total in the ``tag_col``.
+
+    We have records in the rate base table with totals and/or nulls for key tag
+    columns which we want to separate into component parts. This is done in two
+    steps:
+
+    * :func:`get_tag_col_ratio` : for each ``report_year`` and ``utility_id_ferc1``,
+      get a ratio of all of the ``ending_balance`` for all of the non-null and
+      non-total tags.
+    * use this ratio to disaggregate the ``ending_balance`` from records with total
+      and null tags across the component tags.
+
+    Args:
+        rate_base_df: full table of rate base data.
+        tag_col: column with the tags that contains unlabeled values
+            (ex: null or total).
+    """
+    # this works for both the utility_type and in_rate_base tags because
+    # for both tags unlabeled is represented as a null and/or a "total"
+    unlabeled_mask = (rate_base_df[tag_col] == "total") | rate_base_df[tag_col].isnull()
+    ratio_idx = ["report_year", "utility_id_ferc1"]
+    ratio_df = get_tag_col_ratio(
+        # remove the unlabeled records because total is the value
+        # we want to break_down so we can't have it in the columns to sum up
+        rate_base_labeled_df=rate_base_df[~unlabeled_mask],
+        ratio_idx=ratio_idx,
+        tag_col=tag_col,
+    )
+    unlabeled_break_down_df = (
+        pd.merge(
+            rate_base_df[unlabeled_mask],
+            ratio_df,
+            on=ratio_idx,
+            how="left",
+            validate="m:m",
+            suffixes=("_unlabeled", ""),
+        )
+        .assign(
+            ending_balance=lambda x: x[f"ratio_{tag_col}"].fillna(1) * x.ending_balance,
+        )
```

Comment:
ah good question. I'll add a comment about this. na values from this ratio tag column do in effect = 100%. because the ratio is of the non-total or non-null tags. But occasionally there are no non-null or non-null tags. Which results in nulls in these ratio columns during the left merge. We want to preserve the ending balance's for them even if there are no non-null/non-total tags to disaggregate.